api/v1beta1/grafananotificationpolicy_types.go

```diff
@@ -126,6 +129,34 @@ func (r *Route) ToModelRoute() *models.Route {
 	return out
 }
 
+// isMutuallyExclusive checks if a single route satisfies the mutual exclusivity constraint
+func isMutuallyExclusive(r *Route) bool {
```

Comment:
nitpick: could be a member function `func (r *Route) SelectorMutuallyExclusive` or `SpecValid`